contracts/Removal.sol

```diff
@@ -683,8 +783,12 @@ contract Removal is
     bytes memory data
   ) internal virtual override whenNotPaused {
     address market = address(_market);
+    address certificate = address(_certificate);
     bool isToAllowed = to == market ||
-      (to == address(_certificate) || to == address(0));
+      to == certificate ||
+      to == address(0) ||
+      (hasRole({role: CONSIGNOR_ROLE, account: _msgSender()}) &&
+        (to == certificate || hasRole({role: CONSIGNOR_ROLE, account: to})));
```

Comment:
maybe we should change the name of this variable to be `validTransfer` or something because the logic is no longer just about who `to` is?

Reply:
Agreed.  I would be in favor of splitting up this logic so it's easier to test and reason about.